doc/guides/gcov.rst

```diff
@@ -0,0 +1,44 @@
+.. _gcov:
+
+Coverage reports via GCOV
+#########################
+
+Overview
+********
+This feature will allow the extraction of the coverage reports via GCC GCOV tool.
+In zephyr the gcov is made to run without any support for file systems. Therefore the
+entire coverage dump will be coming from the RAM region. This puts a burden on the
+available ram. Therefore this feature is only limited by the RAM size. Therefore this
+is enabled only on qemu emulation for the moment.
+
+Details
+*******
+There are 2 parts to enable this feature. The first is to enable the coverage for the
+device and the second to enable in the test application. As explained earlier the
+code coverage with gcov is a function of RAM available. Therefore ensure that the
+device has enough RAM when enabling the coverage for it. For example a small device
+like frdm_k64f can run a simple test application but the more complex test
+cases which consume more RAM will crash when coverage is enabled.
+
+To enable the device for coverage, select CONFIG_HAS_COVERAGE_SUPPORT
+in the Kconfig.board file.
+
+To report the coverage for the particular test application set CONFIG_COVERAGE=y.
+
+**Steps to generate code coverage reports:**
```

Comment:
Rewrite this to use a heading and code blocks for the examples:
```
Steps to generate code coverage reports
***************************************

1. Build the code with CONFIG_COVERAGE=y::

      $ cmake -DBOARD=mps2_an385 -DCONFIG_COVERAGE=y ..

#. Store the build and QEMU-emulated run output to to a log file::

      $ make run > log.log

#. Generate the gcov gcda files from the log file that was saved::

      $ python3 scripts/gen_gcov_files.py log.log

#. Find the gcov binary placed in the SDK::

      $ find -iregex ".*gcov"

#. Run gcovr to get the reports::

      $ gcovr -r . --html -o gcov_report/coverage.html --html-details --gcov-executable <gcov_path_in_SDK>
```

Shouldn't the gcovr executable be found on the PATH, or via an environment variable (do you really need to use ``find`` to find it)?

Reply:
>Shouldn't the gcovr executable be found on the PATH, or via an environment variable (do you really need to use find to find it)?

This is needed because we need to have a setup which is agnostic of the host gcc version.